core/src/main/java/org/elasticsearch/index/reindex/WorkerBulkByScrollTaskState.java

```diff
@@ -44,6 +45,11 @@
 
     private static final Logger logger = Loggers.getLogger(WorkerBulkByScrollTaskState.class);
 
+    /**
+     * Maximum wait time allowed for throttling.
+     */
+    private static final long MAX_THROTTLE_WAIT_TIME =  TimeUnit.HOURS.toNanos(1);
```

Comment:
I'd probably use a `TimeValue` instead. That way you never have to wonder "is this in nanos or millis?"